app/vmselect/promql/binary_op.go

```diff
@@ -493,7 +493,7 @@ func binaryOpOr(bfa *binaryOpFuncArg) ([]*timeseries, error) {
 
 	for k, tssRight := range mRight {
 		tssLeft := mLeft[k]
-		if tssLeft == nil {
+		if len(removeEmptySeries(tssLeft)) == 0 { // special case where left is empty
```

Comment:
I was checking further about compatibility with `or` operator between Prometheus and VictoriaMetrics. And it looks to me like they still diverge.
For example, I made the following test case [here](https://github.com/prometheus/prometheus/tree/main/promql/promqltest):
```
load 1m
    foo{job="bar", a="a"} 0 0 1 1 0
    foo{job="baz", a="a"} 1 1 1 1 1

eval range from 0 to 4m step 1m (foo{job="bar"} == 0) or on (a) (foo{job="baz"} == 1)
	foo{job="bar", a="a"} 0 0 _ _ 0
	foo{job="baz", a="a"} _ _ 1 1 _
```
This test creates two series with different values. Then, `or` evaluation returns two series in response. It doesn't fill gaps like in left series with right parts, as VictoriaMetrics does. For example, run the following test in VM:
```
(label_set(time() <= 1200, "a", "a", "b", "b1")) or on(a) label_set(time() > 1200, "a", "a", "b", "b2")
```
what result would you expect to get in response?

Reply:
> `vector1 or vector2` results in a vector that contains all original elements (label sets + values) of vector1 and additionally all elements of vector2 which do not have matching label sets in vector1.

Since they have different label sets, I think we should have both of them in response.